src/QAT/qat/purr/compiler/builders.py

```diff
@@ -572,6 +572,7 @@ def measure(
 
         measure_channel = qubit.get_measure_channel()
         acquire_channel = qubit.get_acquire_channel()
+        weights = qubit.measure_acquire.get('weights', None) if qubit.measure_acquire.get('use_weights', False) else None
```

Comment:
not a big deal but this smells like duplicate `None` semantics. Consider the following:

- `None` means do not use weights
- anything else means use weights, and "the thing" as weights

to avoid an additional `use_weights` entry ?

Reply:
Not sure if I follow? Basically there's two ways to turn weights off, if it's set to None or if use_weights is set to False. Weights mirrors what loaded into memory for demolition and 'use_weights' allows you to turn it off and on without loosing the array. I find flexibility when doing calibrations is helpful for people.
Is this what you ment?

Reply:
> without loosing the array

that's the keyword I needed 😄 just wanted to point it out, happy to hear this choice was made deliberately ^^